### ISSUE TITLE
setLayoutProperty

### ISSUE THREAD

**Issue by GD-Aichi** [openlayers/ol-mapbox-style#195]
I add a new [function](https://github.com/GD-Aichi/ol-mapbox-style/commit/acacc373859132f5f97fee82f6dbfe1f0e7dbe79) in my branch.
Now the question is, if you want to have that in your project?

**Comment by ahocevar** [comment]
Thanks for this! I've been planning to do something similar based on the diff utility in https://github.com/mapbox/mapbox-gl-js/tree/master/src/style-spec. That would allow to reflect Mapbox Style document changes instantly.

**Comment by GD-Aichi** [comment]
OK good to know. Is there already a concept of you or have you already found to build on it?

**Comment by ahocevar** [comment]
No, at this point it's just a plan. Any help would be greatly appreciated.

**Comment by GD-Aichi** [comment]
Then I'll wait, but is this feature needed for this version?
I will also need setFilter, which I will then implement the same.

**Comment by ahocevar** [comment]
After a long time, I finally found time to draft out how style updates could work in ol-mapbox-style, making use of Mapbox's diff utility:

The OpenLayers Map instance already has the current Mapbox Style stored in its `mapbox-style` property. This property will also be set on the OpenLayers Layer instances that ol-mapbox-style applies a style to, using `applyStyle()`, `applyBackground()` or `stylefunction()`. When applying a style, the respective functions will register a `change` listener for this property. The listener will run the diff utility and determine changes that need to be made (addition of layers, removal of layers, changes of filters, layout or paint properties, etc.). Based on that, `stylefunction()` will be run again to make changes to a single layer, or, when `olms()` or `apply()` were used, layers can also be removed or added from and to the map.

I don't have a definitive time frame to implement this, but if nothing more urgent appears, I'm hoping to be able to finish it by the end of April.
